File: common/src/main/java/dev/niostone/interstellar/config/PlanetConfig.java
package dev.niostone.interstellar.config;

import java.util.List;

import me.shedaniel.autoconfig.ConfigData;
import me.shedaniel.autoconfig.annotation.Config;

@Config(name = "planet")
public class PlanetConfig implements ConfigData {
    public boolean enable;
    public int dimensionId;
    public String name;
    public int pressure;
    public int radiation;
    public int tier;
    public boolean enableOxygen;
    public boolean enablePressure;
    public boolean enableRadiation;

    public List<GasConfig> gases;
}

File: common/src/main/kotlin/org/stardustmodding/interstellar/impl/config/PlanetConfig.kt
package org.stardustmodding.interstellar.impl.config

import me.shedaniel.autoconfig.ConfigData
import me.shedaniel.autoconfig.annotation.Config
import org.stardustmodding.interstellar.impl.Interstellar

@Config(name = "planet")
class PlanetConfig : ConfigData {
    var enable: Boolean = false
    var dimensionId: Int = -1
    var name: String

    // This is in kPa
    var pressure: Float = 0.0f

    // In μSv/tick
    var radiation: Float = 0.0f
    var tier: Int = 0
    var enableOxygen: Boolean = true
    var enablePressure: Boolean = true
    var enableRadiation: Boolean = true
    var gases: MutableList<GasConfig>

    constructor() : this(Interstellar.MOD_ID + ":null")

    constructor(name: String) {
        this.name = name
        this.gases = mutableListOf()
    }
}
